minichef/contracts/Strategy.sol

```diff
@@ -0,0 +1,458 @@
+        uint256 amount = balanceOfStaked();
+        if (amount > 0) {
+            chef.withdrawAndHarvest(
+                pid, 
+                amount, 
+                address(this)
+            );
+            return balanceOfWant();
+        }
+    }
+
+    /* ========== MUTATIVE FUNCTIONS ========== */
+
+    function prepareReturn(uint256 _debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 _profit,
+            uint256 _loss,
+            uint256 _debtPayment
+        )
+    {
+        // claim our rewards
+        chef.harvest(pid, address(this));
+
+        // if we have emissionToken to sell, then sell some of it
+        uint256 emissionTokenBalance = emissionToken.balanceOf(address(this));
+        if (emissionTokenBalance > 0) {
+            // sell our emissionToken
+            rewardToOptimal();
+        }
+
+        // deposit optimal to synpase curve strategy
+        if (optimal == 0) {
+            uint256 daiBalance = dai.balanceOf(address(this));
+            if (daiBalance > 0) {
+                uint[] memory dataDai = new uint[](4);
+                dataDai[0] = 0;
+                dataDai[1] = daiBalance;
+                dataDai[2] = 0;
+                dataDai[3] = 0;
+                curve.addLiquidity(dataDai, 0, now);
+            }
+        } else if (optimal == 1) {
+            uint256 usdcBalance = usdc.balanceOf(address(this));
+            if (usdcBalance > 0) {
+                uint[] memory dataUSDC = new uint[](4);
+                dataUSDC[0] = 0;
+                dataUSDC[1] = 0;
+                dataUSDC[2] = usdcBalance;
+                dataUSDC[3] = 0;
+                curve.addLiquidity(dataUSDC, 0, now);
+            }
+        } else if (optimal == 2) {
+            uint256 usdtBalance = usdt.balanceOf(address(this));
+            if (usdtBalance > 0) {
+                uint[] memory dataUSDT = new uint[](4);
+                dataUSDT[0] = 0;
+                dataUSDT[1] = 0;
+                dataUSDT[2] = 0;
+                dataUSDT[3] = usdtBalance;
+                curve.addLiquidity(dataUSDT, 0, now);
+            }
+        } else if (optimal == 3) {
+            uint256 nusdBalance = nusd.balanceOf(address(this));
+            if (nusdBalance > 0) {
+                uint[] memory dataNUSD = new uint[](4);
+                dataNUSD[0] = nusdBalance;
+                dataNUSD[1] = 0;
+                dataNUSD[2] = 0;
+                dataNUSD[3] = 0;
+                curve.addLiquidity(dataNUSD, 0, now);
+            }
+        }
+        // get assets balance
+        uint256 assets = estimatedTotalAssets();
+        uint256 wantBal = balanceOfWant();
+
+        uint256 debt = vault.strategies(address(this)).totalDebt;
+        uint256 amountToFree;
+
+        // this would only happen if the chef somehow lost funds or was drained
+        uint256 chefHoldings = want.balanceOf(address(chef));
```

Comment:
Why do we take the chef holdings here? Shouldn't we take the strategy balance? `want.balanceOf(address(this))`?